HealthcareSystem/Backend/Service/RoomAndEquipment/EquipmentService.cs

```diff
@@ -68,7 +68,10 @@ public Equipment GetEquipmentById(int equipmentId)
         public List<Equipment> GetEquipmentByRoomNumber(int roomNumber)
         {
             List<EquipmentInRooms> equipmentInRooms = _equipmentInRoomRepository.GetEquipmentInRoomsByRoomNumber(roomNumber);
-
+            if (equipmentInRooms == null) 
+            {
+                return null;
+            }
```

Comment:
Will this ever be null? Why would the Repository return null instead of an empty list? I don't think this should be happening.

Reply:
Yes, you are right about that, l made mistake, so that part of code actually returned null, but it should not do that.
I have changed it in last commit too.